BxlAsLib.txt

```diff
@@ -0,0 +1,377 @@
+BuildXL as a library - aka. BXLLib.
+
+Problem statement
+-----------------
+- Performance
+  - Graph construction takes a long time due to DScript parsing, type checking, and evaluation.
+  - DScript frontend is not scalable to handle Office multi-architecture (mularchy) builds.
+
+- Frontend usability
+  - CMake, MsBuild frontends create their own adhoc JSON formats of build graphs.

```

Comment:
>JSON [](start = 52, length = 4)

Don't see how that is relevant since their problem is that the engine can be written in other langauge and therefore needs either an IPC channel -or- a dataformat. 

Do you expect the format of these to change i.e. for ninja we will write the binary fragment format in Java?